platform-dom/src/main/java/com/softicar/platform/dom/elements/image/viewer/DomImageViewer.java

```diff
@@ -1,160 +1,124 @@
 package com.softicar.platform.dom.elements.image.viewer;
 
-import com.softicar.platform.common.core.i18n.IDisplayString;
 import com.softicar.platform.common.io.resource.IResource;
 import com.softicar.platform.dom.DomCssClasses;
-import com.softicar.platform.dom.DomImages;
-import com.softicar.platform.dom.DomTestMarker;
 import com.softicar.platform.dom.elements.DomDiv;
-import com.softicar.platform.dom.elements.DomSpan;
-import com.softicar.platform.dom.elements.bar.DomBar;
-import com.softicar.platform.dom.elements.button.DomButton;
 import com.softicar.platform.dom.style.CssStyle;
 import com.softicar.platform.dom.style.ICssLength;
 import java.util.List;
-import java.util.Optional;
 
 public class DomImageViewer extends DomDiv {
 
 	private final List<? extends IResource> images;
-	private final ICssLength width;
-	private final NavigationBar navigationBar;
-	private final ImageDiv imageDiv;
-	private final RotationDiv rotationDiv;
-	private int currentIndex;
+	private final DomImageViewerToolBar toolBar;
+	private final DomImageViewerCanvas canvas;
+	private DomImageViewerZoomLevel zoomLevel;
+	private boolean rotated;
+	private int pageIndex;
 
-	public DomImageViewer(List<? extends IResource> images, ICssLength width) {
+	public DomImageViewer(List<? extends IResource> images) {
 
 		this.images = images;
-		this.width = width;
-		this.navigationBar = new NavigationBar();
-		this.imageDiv = new ImageDiv();
-		this.rotationDiv = new RotationDiv();
-		this.currentIndex = 0;
+		this.toolBar = new DomImageViewerToolBar(this);
+		this.canvas = new DomImageViewerCanvas(this);
+		this.zoomLevel = DomImageViewerZoomLevel.getDefault();
+		this.rotated = false;
+		this.pageIndex = 0;
 
 		addCssClass(DomCssClasses.DOM_IMAGE_VIEWER);
-		appendChild(navigationBar);
-		appendChild(imageDiv);
-		appendChild(rotationDiv);
+
+		appendChild(toolBar);
+		appendChild(canvas);
 
 		refresh();
 	}
 
-	private void nextImage() {
+	// ------------------------------ size ------------------------------ //
 
-		currentIndex++;
-		refresh();
+	public DomImageViewer setWidth(ICssLength width) {
+
+		setStyle(CssStyle.WIDTH, width);
+		return this;
 	}
 
-	private void previousImage() {
+	public DomImageViewer setHeight(ICssLength height) {
 
-		currentIndex--;
-		refresh();
+		setStyle(CssStyle.HEIGHT, height);
+		return this;
 	}
 
-	private void refresh() {
+	// ------------------------------ zoom ------------------------------ //
 
-		navigationBar.refresh();
-		imageDiv.refresh();
+	public DomImageViewerZoomLevel getZoomLevel() {
+
+		return zoomLevel;
 	}
 
-	private class NavigationBar extends DomBar {
+	public void setZoomLevel(DomImageViewerZoomLevel zoomLevel) {
```

Comment:
Not sure where to put this comment... there's a scrolling glitch when using a zoom level below 100:
![Peek 2022-09-06 09-30](https://user-images.githubusercontent.com/15086658/188573994-fcae40f9-84fe-43f6-a4f1-39ccac816508.gif)

Reply:
fixed